### PR TITLE
Fixes missing item in R&D Production Machines

### PR DESCRIPTION
**Pull request by AtnerNT**
As it turns out:
Capture

As a result of this Protolathes and Circuit Imprinters couldn't load their items. This takes The JsonImport out of a task back onto the main thread. 
Also Includes changes to the metafiles of the new textures from the Protolathe PR that weren't properly included.

CL: [Fix] fixed missing items in the R&D machines

### CODE REVIEW

**Comment by AtnerNT** [comment]
*(Alot of lines were changes to correct the indention levels)*

**Comment by corp-0** [comment]
that doesn't look like correct indentation level. If you want to have classes at the outer most indentation level, you can remove the brackets from the ``namespace`` block, I think it works in our current c# version.

**Comment by Copern** [comment]
> that doesn't look like correct indentation level. If you want to have classes at the outer most indentation level, you can remove the brackets from the `namespace` block, I think it works in our current c# version.

Unfortunately, file scoped namespaces aren't available in our version of C#. It's a C# 10 feature. I believe we have C# 8 and most of C# 9 available to us.

**Comment by corp-0** [comment]
Aww damn

File: UnityProject/Assets/Scripts/Systems/Research/Designs.cs [hunk 1]
```diff
@@ -9,152 +9,170 @@
...
-		};
-		public static Dictionary<string,Design> InternalIDSearch;
 
-	}
-	private static void JsonImportInitialization ()
+	public class Designs : MonoBehaviour
 	{
-		Globals.InternalIDSearch =  new Dictionary<string,Design>();
-		foreach (string FilePath in Globals.PathsList) {
-			string json = (Resources.Load (FilePath) as TextAsset).ToString ();
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Systems/Research/Designs.cs:99]: you could just move this outside of the thread task?

File: UnityProject/Assets/Scripts/Systems/Research/Designs.cs [hunk 2]
```diff
@@ -9,152 +9,170 @@
...
+		{
+			if (!(Globals.IsInitialised))
+			{
+				JsonImportInitialization();
+			}
+		}
+		public static class Globals
+		{
+			public static bool IsInitialised = false;
+			public static List<string> PathsList = new List<string> {
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Systems/Research/Designs.cs]: I'm not sure I like this hardcoded list of strings

File: UnityProject/Assets/Scripts/Systems/Research/Designs.cs [hunk 3]
```diff
@@ -9,152 +9,170 @@
...
-					DesignPass.Category = new List<string> ();
-				}
+		void Awake()
+		{
+			if (!(Globals.IsInitialised))
+			{
+				JsonImportInitialization();
+			}
+		}
+		public static class Globals
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Systems/Research/Designs.cs:90]: why do we need a static ''Globals''?

File: UnityProject/Assets/Scripts/Systems/Research/Designs.cs (hunk 2 above)
Comment (AtnerNT) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs]: I've changed it to get the files inside the specified folder. The folder itself is hardcoded to Metadata/Designs. But we shouldn't be putting designs outside that folder as it is so I think its fine.

File: UnityProject/Assets/Scripts/Systems/Research/Designs.cs (hunk 3 above)
Comment (AtnerNT) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs:90]: Ummm- not entirely sure. However this script wasn't written by me and I don't want to interfere with any future plans by deleting it incase it has a planned purpose.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs:90]: ok fair

File: UnityProject/Assets/Scripts/Systems/Research/Designs.cs [hunk 4]
```diff
@@ -4,157 +4,177 @@
...
+			{
+				var JsonTechnologies = JsonConvert.DeserializeObject<List<Dictionary<String, System.Object>>>(json);
+				for (var i = 0; i < JsonTechnologies.Count(); i++)
+				{
+					Design DesignPass = new Design();
+					DesignPass.InternalID = JsonTechnologies[i]["Internal_ID"].ToString();
+					DesignPass.Name = JsonTechnologies[i]["name"].ToString();
+					DesignPass.ItemID = JsonTechnologies[i]["Item_ID"].ToString();
+
+					if (JsonTechnologies[i].ContainsKey("Description"))
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Systems/Research/Designs.cs:111]: is it needed to check for every key individually?

Comment (AtnerNT) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs:111]: I think it's best. If a specific item doesn't have a description set for instance- you don't want that throwing an error stopping every other design from importing as well

Comment (corp-0) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs:111]: There must be a way to make missing keys a null on the deserialized object

Comment (AtnerNT) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs:111]: The mere act of trying to get the key if it doesnt exist causes an NRE and stops it form imporintg- always have to check with .Contains() first

Comment (corp-0) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs:111]: https://docs.microsoft.com/en-us/answers/questions/291110/json-deserialization-behaviour-regarding-missing-p.html

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs:111]: json newtonsoft could also use that idk , I think it's okay to have the checking thing

Comment (corp-0) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs:111]: Any repetitive thing like that is prone to mistakes and it is also annoying to maintain

**Review by corp-0 — CHANGES_REQUESTED** [review]
do you think you could do what I asked in the conversation so we merge this?

File: UnityProject/Assets/Scripts/Systems/Research/Designs.cs (hunk 4 above)
Comment (AtnerNT) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs:111]: I tested it in editor- the checks have to be there- the problem isnt the Json getting a null field- the mere act or trying to get an invalid key from the dictionary causes the erorr and breaks it. It doesn't work without the check. (Getting the wrong key doesn't make the dictionary return null- it just makes it cause an error). Not sure what we are doing different to that link you sent- all I know- it has been tested to not work without the checks

Comment (corp-0) [reply UnityProject/Assets/Scripts/Systems/Research/Designs.cs:111]: I would have to see how you did it, you are probably missing giving the model a default value, but I will merge for now

**Review by corp-0 — APPROVED** [review]

**Pull request merged**